editor/components/post-taxonomies/index.js

```diff
@@ -18,23 +18,26 @@ import HierarchicalTermSelector from './hierarchical-term-selector';
 import FlatTermSelector from './flat-term-selector';
 import { getCurrentPostType } from '../../store/selectors';
 
-export function PostTaxonomies( { postType, taxonomies } ) {
+export function PostTaxonomies( { postType, taxonomies, taxonomyWrapper = identity } ) {
 	const availableTaxonomies = filter( taxonomies.data, ( taxonomy ) => includes( taxonomy.types, postType ) );
-
+	const taxonomyComponents = availableTaxonomies.map( ( taxonomy ) => {
+		const TaxonomyComponent = taxonomy.hierarchical ? HierarchicalTermSelector : FlatTermSelector;
+		return taxonomyWrapper(
+			<TaxonomyComponent
+				key={ taxonomy.slug }
+				restBase={ taxonomy.rest_base }
+				slug={ taxonomy.slug }
+			/>,
+			taxonomy
+		);
+	} );
+
+	// Fragment is used because enzyme shallow does not seems to handle correctly components returning just arrays.
```

Comment:
Are you sure it's not just that `shallow` won't traverse into the `children` ? Have you tried with `mount` instead?

Reply:
Not sure of the reason but all my tries to test returning array failed and I observed in that debug method returned totally invalid data. At the time I found a recent PR on enzyme repository, related to this https://github.com/airbnb/enzyme/pull/1498, maybe these changes are not yet available in our version.

I also tried to use mount instead, but unfortunately, it was not possible because of errors related with our withAPIData HOC I think we would need to mock it to use mount here, which was something unnecessary for testing purposes of this component.

Reply:
It's very unintuitive, but this seemed to work okay for me:

```diff
diff --git a/editor/components/post-taxonomies/index.js b/editor/components/post-taxonomies/index.js
index d7f28db94..200d61f06 100644
--- a/editor/components/post-taxonomies/index.js
+++ b/editor/components/post-taxonomies/index.js
@@ -8,7 +8,7 @@ import { filter, identity, includes } from 'lodash';
  * WordPress dependencies
  */
 import { withAPIData } from '@wordpress/components';
-import { compose, Fragment } from '@wordpress/element';
+import { compose } from '@wordpress/element';
 
 /**
  * Internal dependencies
@@ -20,7 +20,8 @@ import { getCurrentPostType } from '../../store/selectors';
 
 export function PostTaxonomies( { postType, taxonomies, taxonomyWrapper = identity } ) {
 	const availableTaxonomies = filter( taxonomies.data, ( taxonomy ) => includes( taxonomy.types, postType ) );
-	const taxonomyComponents = availableTaxonomies.map( ( taxonomy ) => {
+
+	return availableTaxonomies.map( ( taxonomy ) => {
 		const TaxonomyComponent = taxonomy.hierarchical ? HierarchicalTermSelector : FlatTermSelector;
 		return taxonomyWrapper(
 			<TaxonomyComponent
@@ -31,14 +32,6 @@ export function PostTaxonomies( { postType, taxonomies, taxonomyWrapper = identi
 			taxonomy
 		);
 	} );
-
-	// Fragment is used because enzyme shallow does not seems to handle correctly components returning just arrays.
-	// Using debug method it was possible to verify shallow always output <undefined /> when arrays are used.
-	return (
-		<Fragment>
-			{ taxonomyComponents }
-		</Fragment>
-	);
 }
 
 const applyConnect = connect(
diff --git a/editor/components/post-taxonomies/test/index.js b/editor/components/post-taxonomies/test/index.js
index 42cfed70f..1f21845a1 100644
--- a/editor/components/post-taxonomies/test/index.js
+++ b/editor/components/post-taxonomies/test/index.js
@@ -16,7 +16,7 @@ describe( 'PostTaxonomies', () => {
 			<PostTaxonomies postType="page" taxonomies={ taxonomies } />
 		);
 
-		expect( wrapper.children() ).toHaveLength( 0 );
+		expect( wrapper.at( 0 ) ).toHaveLength( 0 );
 	} );
 
 	it( 'should render taxonomy components for taxonomies assigned to post type', () => {
@@ -43,8 +43,8 @@ describe( 'PostTaxonomies', () => {
 			<PostTaxonomies postType="page" taxonomies={ taxonomies } />
 		);
 
-		expect( wrapper.children() ).toHaveLength( 1 );
-		expect( wrapper.childAt( 0 ).props() ).toEqual( {
+		expect( wrapper.at( 0 ) ).toHaveLength( 1 );
+		expect( wrapper.at( 0 ).at( 0 ).props() ).toEqual( {
 			slug: 'category',
 			restBase: 'categories',
 		} );
```

https://github.com/airbnb/enzyme/issues/1149 is the (open) parent issue for array-returning components in Enzyme.